herd/libdir/asl.cat

```diff
@@ -62,12 +61,15 @@ let restrict-finals s =
 let AArch64 = NASLLocal
 let AArch64Finals =  restrict-finals(AArch64&Wreg)
 
-(* DATA and NDATA *)
-let to_data = (asl_deps_restricted ; [DATA] ; asl_iico_data ; [W | Wreg]) & aarch64
+(* Compute AArch64 ADDR *)
+let to_addr = (asl_deps_restricted ; [ADDR] ; asl_iico_data ) & aarch64
 
-let AArch64_DATA = NASLLocal & domain (to_data)
-show [AArch64_DATA] as AArch64_DATA
-show [DATA] as debug_data
+let AArch64_ADDR = NASLLocal & domain (to_addr)
+let to_memwrite = aarch64_iico_data;aarch64_iico_ctrl?;[W]
+let AArch64_DATA = ((NASLLocal\M) & domain (to_memwrite)) \ AArch64_ADDR
```

Comment:
Data ports are computed here. I considered more convenient to use the aarch64_iico_... relations, as they appear on the final diagrams. The `aarch64_iico_ctrl?` component is necessary to include the two register read effects into the `DATA` set.